vmm/src/vm.rs

```diff
@@ -385,17 +387,17 @@ impl Vm {
 
         let cmdline_cstring = CString::new(cmdline).map_err(Error::CmdLineCString)?;
         let guest_memory = self.memory_manager.lock().as_ref().unwrap().guest_memory();
-        let mem = guest_memory.load_full();
+        let mem = guest_memory.memory();
```

Comment:
Instead of `load_full()` you can use `memory().into_inner()`. If this is not needed, it should be possible to use `&*mem` instead of `mem.deref()`.

Reply:
@rbradford @bonzini not sure if we need `into_inner()` here since this means we're going to get a clone, but it isn't really needed at this point in the code, is it?
Second thing, about the `Deref`, I thought it was cleaner for me to use the `deref()` implementation rather than using `&*mem`. What's the best practice and why?

Reply:
No it's not needed I think, unless you want a 1:1 conversion. Regarding `deref()` I'm too n00b to give an answer. I guess it's a matter of taste, but there is also a [clippy issue about it](https://github.com/rust-lang/rust-clippy/issues/1566) so I guess `&*x` is preferrable even though it looks like Perl.